File: python/scripts/live_smoke.py
from __future__ import annotations

import os
import sys

from buble_ai import Buble, BubleAPIError


def check(condition: bool, message: str) -> None:
    if not condition:
        raise AssertionError(message)


def main() -> int:
    if not os.environ.get("BUBLE_API_KEY"):
        print("Missing BUBLE_API_KEY.", file=sys.stderr)
        return 1

    client = Buble()

    models = client.media_models.list()
    check(isinstance(models.get("data"), list), "media_models.list() should return data list")
    print(f"PASS media_models.list(): {len(models['data'])} models")

    image_models = client.media_models.list(media_type="image")
    check(isinstance(image_models.get("data"), list), "filtered media_models.list() should return data list")
    print(f"PASS media_models.list(media_type='image'): {len(image_models['data'])} models")

    apps = client.apps.list(limit=10)
    check(isinstance(apps.get("data"), list), "apps.list() should return data list")
    print(f"PASS apps.list(): {len(apps['data'])} apps")

    if apps["data"]:
        app_id = apps["data"][0]["id"]
        app = client.apps.retrieve(app_id)
        check(app["data"]["id"] == app_id, "apps.retrieve() should return requested id")
        print(f"PASS apps.retrieve({app_id})")

    chat_models = client.chat.models.list()
    check(chat_models.get("object") == "list", "chat.models.list() should preserve object=list")
    check(isinstance(chat_models.get("data"), list), "chat.models.list() should return data list")
    print(f"PASS chat.models.list(): {len(chat_models['data'])} models")

    try:
        client.generations.retrieve("sdk-smoke-non-existent-task")
    except BubleAPIError as error:
        check(error.status_code >= 400, "expected API error for non-existent task")
        print(f"PASS BubleAPIError parsing: {error.status_code} {error.code or ''}".strip())
    else:
        raise AssertionError("non-existent task should not succeed")

    client.close()
    print("PASS live smoke test completed without creating billable generation tasks")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: npm/scripts/live-smoke.mjs
#!/usr/bin/env node
import { Buble, BubleAPIError } from '../dist/esm/index.js';

const apiKey = process.env.BUBLE_API_KEY;
const baseURL = process.env.BUBLE_BASE_URL || 'https://buble.ai';

if (!apiKey) {
  console.error('Missing BUBLE_API_KEY. Run with BUBLE_API_KEY=sk_...');
  process.exit(1);
}

function pass(message) {
  console.log(`PASS ${message}`);
}

function info(message) {
  console.log(`INFO ${message}`);
}

function assert(condition, message) {
  if (!condition) {
    throw new Error(message);
  }
}

async function main() {
  const buble = new Buble({
    apiKey,
    baseURL,
    timeout: 30_000
  });

  info(`Base URL: ${baseURL}`);

  const mediaModels = await buble.mediaModels.list();
  assert(Array.isArray(mediaModels.data), 'mediaModels.list() should return { data: array }.');
  assert(mediaModels.data.length > 0, 'mediaModels.list() returned no models.');
  pass(`mediaModels.list(): ${mediaModels.data.length} models`);

  const imageModels = await buble.mediaModels.list({ media_type: 'image' });
  assert(Array.isArray(imageModels.data), 'mediaModels.list({ media_type }) should return { data: array }.');
  pass(`mediaModels.list({ media_type: image }): ${imageModels.data.length} models`);

  const apps = await buble.apps.list({ limit: 10 });
  assert(Array.isArray(apps.data), 'apps.list() should return { data: array }.');
  pass(`apps.list(): ${apps.data.length} apps in first page`);

  if (apps.data[0]?.id) {
    const app = await buble.apps.retrieve(apps.data[0].id);
    assert(app.data.id === apps.data[0].id, 'apps.retrieve() should return the requested app id.');
    assert(Array.isArray(app.data.input_parameters), 'apps.retrieve() should expose input_parameters.');
    pass(`apps.retrieve(${apps.data[0].id})`);
  }

  const chatModels = await buble.chat.models.list();
  assert(chatModels.object === 'list', 'chat.models.list() should preserve OpenAI-style object=list response.');
  assert(Array.isArray(chatModels.data), 'chat.models.list() should return data array.');
  assert(chatModels.data.length > 0, 'chat.models.list() returned no chat models.');
  pass(`chat.models.list(): ${chatModels.data.length} models`);

  try {
    await buble.generations.retrieve('sdk-smoke-non-existent-task');
  } catch (error) {
    if (error instanceof BubleAPIError) {
      assert(error.status >= 400, 'retrieve(non-existent) should return an API error status.');
      pass(`BubleAPIError parsing: ${error.status}${error.code ? ` ${error.code}` : ''}`);
    } else {
      throw error;
    }
  }

  pass('live smoke test completed without creating billable generation tasks');
}

main().catch((error) => {
  console.error('FAIL live smoke test failed');
  if (error instanceof BubleAPIError) {
    console.error(`API ${error.status}${error.code ? ` ${error.code}` : ''}: ${error.message}`);
    if (error.details) console.error(JSON.stringify(error.details, null, 2));
  } else {
    console.error(error);
  }
  process.exit(1);
});
